blogposts/2021/integration-testing.md

```diff
@@ -0,0 +1,115 @@
+---
+title: 'How we added integration testing to our CI pipeline'
+externalTitle: 'Adding integration testing to our CI'
+description: 'Here's the story, and the lessons learned, from our work to remove all existing backend-related end-to-end tests and reliably run their corresponding unit and/or integration tests as part of our CI pipeline on all branches. 
+externalDescription: 'Integration testing isn't glamorous work, but it is impactful. Here's a behind-the-scenes look at how we worked toward a more reliable, less flaky testing suite.'
+author: Joe Chen
+publishDate: 2021-12-22T18:00+02:00
+tags: [blog]
+slug: integration-testing
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/how-not-to-break-a-search-engine-unglamorous-engineering.jpg
+published: true
+---
+
+![Unglamorous engineering graphic](https://storage.googleapis.com/sourcegraph-assets/blog/how-not-to-break-a-search-engine-unglamorous-engineering.jpg)
+
+# How we added integration testing to our CI pipeline
+
+My name is Joe Chen and I’m on the Core Application team here at Sourcegraph. It’s our responsibility to cover all the needs of the other teams. Some of our previous work includes licensing enforcement, background permissions syncing, and an explicit permissions API.
```

Comment:
@unknwon Could we clarify what we mean here by "all the needs"? Can we specify, like, would saying "infrastructure needs" be accurate?

Reply:
>Core Application team

This team no longer exists.

```suggestion
My name is Joe Chen and I was on the Core Application team at Sourcegraph. Our responsibility was to build and maintain the infrastructure of the Sourcegraph application for other teams. Some of our previous work includes licensing enforcement, background permissions syncing, and an explicit permissions API.
```

Currently I'm on Cloud SaaS but on a different mission, so I think current tense is not accurate. Feel free to tweak my suggestion before applying!